tests/wol/test_wol.py

```diff
@@ -0,0 +1,387 @@
+        pytest_assert(pkt.type == ETHER_TYPE_WOL_DEC)
+        pytest_assert(pkt.load == build_magic_packet_payload(target_mac))
+
+    with capture_and_check_packet_on_dut(
+        duthost=ptfhost,
+        interface='eth'+str(randon_ptf_port),
+        pkts_filter=WOL_ETHER_PKT_FILTER,
+        pkts_validator=validate_wol_broadcast_port
+    ):
+        duthost.shell("sudo wol %s %s -b" % (random_dut_port, target_mac))
+
+
+@pytest.mark.parametrize("password", ["11:22:33:44:55:66", "192.168.0.1"])
+def test_send_with_password(
+    duthost,
+    ptfhost,
+    get_connected_dut_intf_to_ptf_index,
+    password
+):
+    dut_mac = duthost.facts['router_mac']
+    target_mac = "1a:2b:3c:d1:e2:f3"
+    connected_dut_intf_to_ptf_index = get_connected_dut_intf_to_ptf_index
+    random_dut_port, randon_ptf_port = random.choice(connected_dut_intf_to_ptf_index)
+    logging.info("Test with random dut port %s and ptf port index %s" % (random_dut_port, randon_ptf_port))
+
+    def validate_wol_broadcast_port(pkts):
+        pytest_assert(len(pkts) == 1, "Unexpected pkts count %s" % len(pkts))
+        pkt = pkts[0]
+        pytest_assert(pkt.dst == target_mac, "Unexpected dst mac %s" % pkt.dst)
+        pytest_assert(pkt.src == dut_mac, "Unexpected src mac %s" % pkt.src)
+        pytest_assert(pkt.type == ETHER_TYPE_WOL_DEC)
+        pytest_assert(pkt.load == build_magic_packet_payload(target_mac, password))
+
+    with capture_and_check_packet_on_dut(
+        duthost=ptfhost,
+        interface='eth'+str(randon_ptf_port),
+        pkts_filter=WOL_ETHER_PKT_FILTER,
+        pkts_validator=validate_wol_broadcast_port
+    ):
+        duthost.shell("sudo wol %s %s -p %s" % (random_dut_port, target_mac, password))
+
+
+@pytest.mark.parametrize("interval", [0, 2000])
+@pytest.mark.parametrize("count", [2, 5])
+def test_single_interface_with_count_and_interval(
+    duthost,
+    ptfhost,
+    get_connected_dut_intf_to_ptf_index,
+    interval,
+    count
+):
+    dut_mac = duthost.facts['router_mac']
+    target_mac = "1a:2b:3c:d1:e2:f4"
+    connected_dut_intf_to_ptf_index = get_connected_dut_intf_to_ptf_index
+    random_dut_port, randon_ptf_port = random.choice(connected_dut_intf_to_ptf_index)
+    logging.info("Test with random dut port %s and ptf port index %s" % (random_dut_port, randon_ptf_port))
+
+    def validate_wol_pkts(pkts):
+        pytest_assert(len(pkts) == count, "Unexpected pkts count %s" % len(pkts))
+        last_time = None
+        for pkt in pkts:
+            pytest_assert(pkt.dst == target_mac, "Unexpected dst mac %s" % pkt.dst)
+            pytest_assert(pkt.src == dut_mac, "Unexpected src mac %s" % pkt.src)
+            pytest_assert(pkt.type == ETHER_TYPE_WOL_DEC)
+            pytest_assert(pkt.load == build_magic_packet_payload(target_mac))
+            if last_time:
+                millseconds_gap = (pkt.time - last_time) * 1000
+                pytest_assert(millseconds_gap >= interval and millseconds_gap < interval + 5,
```

Comment:
The real time interval between two packets must be greater or equal to the interval provided.

Because in both versions of wol, the interval is the time between the last packet sent and current packet ready to send.

So current code works fine.